src/System Application/App/Agent/Setup/AgentImpl.Codeunit.al

```diff
@@ -425,6 +426,41 @@ codeunit 4301 "Agent Impl."
         Hyperlink(NoAgentsAvailableNotificationLearnMoreUrlLbl);
     end;
 
+    local procedure GenerateUniqueUserName(AgentUserName: Code[50]): Code[50]
+    var
+        User: Record User;
+        UniqueUserName: Text[50];
+        AgentNamePrefix: Text[50];
+        NumberOfAgentDigits: Integer;
+        MaximumPrefixLength: Integer;
+        AgentNumberSeparatorTok: Label '-', Locked = true;
+    begin
+        // Check if the user name is already unique
+        User.Ascending(false);
+        User.SetRange("User Name", AgentUserName);
+        User.ReadIsolation := IsolationLevel::ReadUncommitted;
+        if User.IsEmpty() then
+            exit(AgentUserName);
+
+        // If not check if there is a user with digits at the end of the name
+        NumberOfAgentDigits := 2;
```

Comment:
Edge case: We have environments with more than 100 companies, this will result in the same issue if they enable the same agent on those companies.